clients/unshaded/src/main/java/tachyon/client/lineage/LineageMasterClient.java

```diff
@@ -0,0 +1,159 @@
+   */
+  public LineageMasterClient(InetSocketAddress masterAddress, ExecutorService executorService,
+      TachyonConf tachyonConf) {
+    super(masterAddress, executorService, tachyonConf);
+  }
+
+  @Override
+  protected String getServiceName() {
+    return Constants.LINEAGE_MASTER_SERVICE_NAME;
+  }
+
+  @Override
+  protected void afterConnect() {
+    mClient = new LineageMasterService.Client(mProtocol);
+  }
+
+  public synchronized long createLineage(List<TachyonURI> inputFiles, List<TachyonURI> outputFiles,
+      CommandLineJob job) throws IOException, FileDoesNotExistException {
+    // prepare for RPC
+    List<String> inputFileStrings = Lists.newArrayList();
+    for (TachyonURI inputFile : inputFiles) {
+      inputFileStrings.add(inputFile.toString());
+    }
+    List<String> outputFileStrings = Lists.newArrayList();
+    for (TachyonURI outputFile : outputFiles) {
+      outputFileStrings.add(outputFile.toString());
+    }
+
+    int retry = 0;
+    while (!mClosed && (retry ++) <= RPC_MAX_NUM_RETRY) {
+      connect();
+      try {
+        return mClient.createLineage(inputFileStrings, outputFileStrings,
+            job.generateCommandLineJobInfo());
+      } catch (FileDoesNotExistException e) {
+        throw e;
+      } catch (TException e) {
+        LOG.error(e.getMessage(), e);
+        mConnected = false;
+      }
+    }
+    throw new IOException("Failed after " + retry + " retries.");
+  }
+
+  public synchronized boolean deleteLineage(long lineageId, boolean cascade)
+      throws IOException, LineageDoesNotExistException, LineageDeletionException {
+    int retry = 0;
+    while (!mClosed && (retry ++) <= RPC_MAX_NUM_RETRY) {
+      connect();
+      try {
+        return mClient.deleteLineage(lineageId, cascade);
+      } catch (TException e) {
+        LOG.error(e.getMessage(), e);
+        mConnected = false;
+      }
+    }
+    throw new IOException("Failed after " + retry + " retries.");
+  }
+
+  public synchronized long recreateFile(String path, long blockSizeBytes, long ttl)
+      throws IOException, LineageDoesNotExistException {
+    int retry = 0;
+    while (!mClosed && (retry ++) <= RPC_MAX_NUM_RETRY) {
+      connect();
+      try {
+        return mClient.recreateFile(path, blockSizeBytes, ttl);
+      } catch (TException e) {
+        LOG.error(e.getMessage(), e);
+        mConnected = false;
+      }
+    }
+    throw new IOException("Failed after " + retry + " retries.");
+  }
+
+  public synchronized void asyncCompleteFile(long fileId) throws IOException {
+    int retry = 0;
+    while (!mClosed && (retry ++) <= RPC_MAX_NUM_RETRY) {
+      connect();
+      try {
+        mClient.asyncCompleteFile(fileId);
+        return;
+      } catch (TException e) {
+        LOG.error(e.getMessage(), e);
+        mConnected = false;
+      }
+    }
+    throw new IOException("Failed after " + retry + " retries.");
+  }
+
+  public synchronized List<LineageInfo> listLineages() throws IOException {
```

Comment:
Sure. I used it because I think `listLineages` is shorter in the shell. But for RPC we can use `getLineageInfoList()` for consistency.